sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/azure/fix.adoc

```diff
@@ -0,0 +1,36 @@
+== Why is this an issue?
+
+Hardcoding secret is one of the most widespread causes of secret leaks. It happens
+when a developper or a machine writes a password into a file that is pushed into
+the code base afterwards.
+
+In general, passwords, secrets, and any type of credentials should only be used to
+authenticate a single entity (a person or a system).
+
+Keeping credentials in plain text in a code base is tantamount to sharing that
+password with anyone who has access to the source code and runtime servers. +
+Thus, it is a breach of trust, as these individuals have the ability to
+impersonate others.
+
+Most of the time, as leaking a secret allows third parties to authenticate as
+another system or person, they can impersonate legitimate identities and
+undermine trust within the organization. Third-parties include other
+developpers, end-users, bots that crawl git repositories or malicious actors
+after penetrating the organization. +
+It does not matter if the impersonation is malicious: In either case, it is a
+clear breach of trust in the system, as the systems involved falsely assume
+that the authenticated entity is who it claims to be. +
+The consequences can be catastrophic.
+
+Only administrators should have access to storage account keys. To authorize an
+application to access an Azure Storage, it is recommended to create a service
+principal and assign it the required privileges only.
+
+https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory?toc=%2Fazure%2Fstorage%2Fblobs%2Ftoc.json#authenticate-with-the-azure-identity-library[The Azure Identity SDK]
+provides several options such as DefaultAzureCredential that can be used to
+retrieve secrets from, for instance, environment variables.
+
+Storage account keys should not be stored with the application code or saved
+anywhere in plain text accessible to others. Consider using an
+https://azure.microsoft.com/en-us/products/key-vault/[Azure Key Vault] to store
+and manage keys.
```

Comment:
@hendrik-buchwald-sonarsource do you think we should make a non-generic text and keep this azure-specific text ? I added it because it was already there in the existing rule but I am not sure if it's worth keeping it